huntsman_dust/util_fake_image.py

```diff
@@ -0,0 +1,144 @@
+                         random_state=random_state)
+    x_dev = powerlaw.rvs(1/2,
+                         loc=std_dev[0],
+                         scale=std_dev[1] - std_dev[0],
+                         size=n_sources,
+                         random_state=12345)
+    y_dev = powerlaw.rvs(1/2,
+                         loc=std_dev[0],
+                         scale=std_dev[1] - std_dev[0],
+                         size=n_sources,
+                         random_state=12346)
+    amplitude = Column(amp_p, name='amplitude')
+    x_stddev = Column(x_dev, name='x_stddev')
+    y_stddev = Column(y_dev, name='y_stddev')
+    sources.add_column(amplitude, index=0)
+    sources.add_column(x_stddev, index=3)
+    sources.add_column(y_stddev, index=4)
+
+    if noise is None:
+        image = make_gaussian_sources_image(shape, sources)
+    else:
+        image = image + make_noise_image(shape,
+                                         type=noise,
+                                         mean=noise_mean,
+                                         stddev=noise_stddev)
+
+    # CREATING BACKGROUNG (ISM)
+    # The objective is to create a background with different levels of
+    # spatial fluctuations built in. This is achived by first creating the
+    # desired 2D power spectrum, which is a radial power law in Fourier space.
+    # According to litrature, the index of ISM power law distribution is -2.9
+    # Taking the inverse FFT of this p_law array gives a background with the
+    # desired levels of spatial fluctuations in real space.
+
+    p_law = np.zeros(shape, dtype=float)
+    y, x = np.indices(p_law.shape)
+    center = np.array([(y.max()-y.min())/2.0, (x.max()-x.min())/2.0])
+    r = np.hypot(x - center[1], y - center[0])
+
+    r_ind = r.astype(int)
+    r_max = r.max().astype(int)
+
+    a = np.arange(2, r_max+3, 1)      # These values control size of clouds
+    b = 10**14*a**(-2.9)              # This controls magnitude of background
+
+    for i in range(0, r_max+1):
+        p_law[r_ind > i] = b[i]
+
+    magnitude = np.sqrt(p_law)
+    phase = 2*np.pi*np.random.randn(512, 512)
+    FFT = magnitude * np.exp(1j*phase)
+    FFT2 = fftpack.fftshift(FFT)
+    FFT1 = (fftpack.ifft2(FFT2))
+    background = np.abs(FFT1)
+
+    utils.plt_image(image,
+                    figure=1,
+                    title="Gaussian Sources Image",
+                    xlabel="Pixels",
+                    ylabel="Pixels",
+                    cmap='Greys_r')
+
+    utils.plt_image(background,
+                    figure=2,
+                    title="Background",
+                    xlabel="Pixels",
+                    ylabel="Pixels",
+                    cmap='Greys_r')
+
+    utils.plt_image(image+background,
+                    figure=3,
+                    title="Fake Image",
+                    xlabel="Pixels",
+                    ylabel="Pixels",
+                    cmap='Greys_r')
+
+    plt.show()
```

Comment:
I think you definitely want capacity to write-out the file to a new fits file so you can see it in DS9. So perhaps request an optional parameter like: `output_fits_filename=None` and if not None, then write out to the location. Same for plots too -- they are handy to have a hard copy for, e.g. `output_plot_filename`. 

And eventually plotting itself will be optional or a separate function as a user might want to just run it without plotting. Run it a few times and then plot one., etc. More flexible if they are separated. But this can happen later...